com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -258,6 +258,21 @@ public boolean visit(LambdaExpression node) {
                 }
                 return super.visit(node);
             }
+
+            private int findNearestRelatedLineToLambda(LambdaExpression lambda) {
+                ASTNode node = lambda;
+                while (node != null) {
+                    int line = astUnit.getLineNumber(node.getStartPosition());
+                    if(line == sourceLine) {
+                        return line;
+                    } else if (line < sourceLine) {
+                        // the lambda doesn't belong to current line at all
+                        break;
+                    }
+                    node = node.getParent();
```

Comment:
Recursively checking the parent is not efficient, any way to return earlier?

Reply:
Well do you see that we could run forever where the`line < sourceLine` will be never reached ? Otherwise i think in most cases we will end from the first iteration where `line == sourceLine` right ? Only in a multiline scenario we will traverse to see if the lambda we found belongs to the same line we add a breakpoint. And I also assume this logic will be executed only for breakpoints that are installed on the source line isn't it ?

Reply:
Updated the fix